app/experimenter/experiments/serializers.py

```diff
@@ -356,10 +379,16 @@ def get_filter_object(self, obj):
         return filter_objects
 
     def get_arguments(self, obj):
+        min_version = obj.firefox_min_version_integer
         if obj.is_pref_experiment:
             return ExperimentRecipePrefArgumentsSerializer(obj).data
-        elif obj.is_addon_experiment:
+        elif (
+            obj.is_addon_experiment
+            and min_version <= obj.firefox_min_addon_serializer_version
```

Comment:
How about `obj.is_branched_addon` method on the model.....? 🤔

Reply:
I imagine we'll need to do this check more than once....

Reply:
oooh you're right, sounds good will do